pandas/tests/interchange/test_impl.py

```diff
@@ -176,3 +176,13 @@ def test_nonstring_object():
     col = df.__dataframe__().get_column_by_name("A")
     with pytest.raises(NotImplementedError, match="not supported yet"):
         col.dtype
+
+
+def test_datetime():
+    df = pd.DataFrame({"A": [pd.Timestamp("2022-01-01"), pd.NaT]})
+    col = df.__dataframe__().get_column_by_name("A")
+
+    assert col.size == 2
+    assert col.null_count == 1
+    assert col.dtype[0] == DtypeKind.DATETIME
+    assert col.describe_null == (ColumnNullType.USE_SENTINEL, pd.NaT)
```

Comment:
Can you also test the roundtrip here to ensure we correctly handle it in from_dataframe? (unless that is already tested elsewhere for NaT?)

(although we currently ignore the describe_null indication for datetime anyway .. Which we maybe should update to error if it is anything else?)

Reply:
Sure, added the roundtrip test for this dataframe with `pd.NaT`.

Currently, looks like we raise `NotImplementedError` if the the null type is not `USE_SENTINEL, USE_BITMASK, USE_BYTEMASK, NON_NULLABLE, or USE_NAN`